recipes/libvault/all/patches/fix-cmake-0.48.0.patch

```diff
@@ -0,0 +1,80 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -4,7 +4,6 @@
+ set(CMAKE_CXX_STANDARD 17)
+ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ set(CMAKE_CXX_EXTENSIONS OFF)
+-set(CMAKE_CXX_FLAGS_DEBUG --coverage)
+ set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+ option(ENABLE_TEST "Enable tests?" ON)
+@@ -14,13 +13,6 @@
+ option(BUILD_SHARED_LIBS "Build vault as a shared library" ON)
+ option(INSTALL "Run install targets" ON)
+
+-find_package(CURL)
+-if (CURL_FOUND)
+-  include_directories(${CURL_INCLUDE_DIR})
+-else (CURL_FOUND)
+-  message(FATAL_ERROR "CURL not found")
+-endif (CURL_FOUND)
+-
+ include(GNUInstallDirs)
+ include_directories("${PROJECT_SOURCE_DIR}/lib")
+
+@@ -120,11 +112,20 @@
+
+ target_include_directories(vault PRIVATE src)
+
++OPTION(UseCurl "UseCurl" ON)
++IF (UseCurl)
++    FIND_PACKAGE(CURL)
++    IF (CURL_FOUND)
++        target_link_libraries(vault CURL::libcurl)
++    ENDIF()
++ENDIF()
++
+ if(LINK_CURL)
+   target_link_libraries(vault curl)
+ endif(LINK_CURL)
+
+ if (ENABLE_COVERAGE)
++  set(CMAKE_CXX_FLAGS_DEBUG --coverage)
+   target_link_libraries(vault gcov)
+ endif ()
+
+@@ -150,25 +151,9 @@
+   write_basic_package_version_file(
+           ${CMAKE_CURRENT_BINARY_DIR}/libvaultConfigVersion.cmake
+           COMPATIBILITY SameMajorVersion)
+-
+-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfig.cmake"
+-          "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfigVersion.cmake"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/libvault")
+-
+-  configure_file(vault.pc.in vault.pc @ONLY)
+-  install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+ endif (INSTALL)
+
+ if (ENABLE_TEST)
+-  include(FetchContent)
+-  FetchContent_Declare(
+-          Catch2
+-          GIT_SHALLOW    TRUE
+-          GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+-          GIT_TAG        v2.13.6)
+-  FetchContent_MakeAvailable(Catch2)
+-  list(APPEND CMAKE_MODULE_PATH ${Catch2_SOURCE_DIR}/contrib)
+   enable_testing()
```

Comment:
```suggestion
           COMPATIBILITY SameMajorVersion)
-
-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfig.cmake"
-          "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfigVersion.cmake"
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/libvault")
-
-  configure_file(vault.pc.in vault.pc @ONLY)
-  install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc"
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 endif (INSTALL)

 if (ENABLE_TEST)
-  include(FetchContent)
-  FetchContent_Declare(
-          Catch2
-          GIT_SHALLOW    TRUE
-          GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-          GIT_TAG        v2.13.6)
-  FetchContent_MakeAvailable(Catch2)
-  list(APPEND CMAKE_MODULE_PATH ${Catch2_SOURCE_DIR}/contrib)
   enable_testing()
```

Don't need to patch this part, you can use tools.rmdir as other recipes and remove the produced cmake folder.

Reply:
changed